docs/content/manual/manual.md

```diff
@@ -0,0 +1,5538 @@
+Thus `as` functions as something of a foreach loop.
+
+You can write `{$foo}` as shorthand for `{foo: $foo}`.
+
+::: Examples
+
+~~~
+.bar as $x | .foo | . + $x
+{"foo":10, "bar":200}
+210
+~~~
+
+~~~
+. as $i|[(.*2|. as $i| $i), $i]
+5
+[10,5]
+~~~
+
+:::
+
+## Scoping
+
+There are three types of symbols in jq:
+[variables](#variables),
+[labels](#label-break), and
+[functions](#function-definitions).
+All of these symbols are scoped lexically, with filters being able to refer only to
+symbols that have been defined "to the left" of them.
+Furthermore, there is no way to change the value of a binding;
+one can only create a new binding with the same name,
+but this will not be visible where the old one was.
+
+::: Example
+In the filter
+
+    .realnames as $names | (.posts[] | {title, author: $names[.author]})
+
+the binding `$names` is visible "to the right" of it, but in the filter
+
+    (.realnames as $names | .posts[]) | {title, author: $names[.author]}
+
+the binding `$names` is _not_ visible past the closing parenthesis,
+so the filter is not well-formed.
+:::
+
+::: Example
+The filter `1 as $x | (2 as $x | $x), $x` returns the values `2, 1`.
+First,
+it introduces a variable `$x` via `1 as $x`, then
+it introduces a variable `$x` via `2 as $x`  that _shadows_ the previous `$x`.
+However, because we limit the scope of `2 as $x` with parentheses,
+the final `$x` refers to the original `1 as $x` again.
+:::
+
+::: Note
+Labels and variables look alike, yet they live in different worlds.
+To see this, consider the filter `1 as $x | label $x | $x, break $x, 2`.
+If the variable `$x` and the label `$x` would live in the same world,
+then the label `$x` would shadow the variable `$x`.
+However, because they live in different worlds,
+they do not shadow each other, therefore
+this filter is syntactically correct and returns `1`.
+:::
+
+## Destructuring
+
+So far in this section, every variable binding
+`f as $x | g` bound exactly one variable `$x` in `g`.
+Now, we will introduce a mechanism to bind _multiple_ variables in one `as` binding.
+For this, we write `f as p | g`, where `p` is a
+_pattern_ that matches the structure of the outputs of `f`.
+
+::: Example
+
+The filter
+
+    . as {realnames: $names, posts: [$first, $second]} | g
+
+is equivalent to the filter
+
+    .realnames as $names  |
+    .posts[0]  as $first  |
+    .posts[1]  as $second |
+    g
+
+:::
+
+The variable declarations in array patterns (e.g., `. as [$first, $second]`)
+bind the elements of the array from the element at index zero on up, in order.
+When there is no value at the index for an array pattern element,
+`null` is bound to that variable.
+
+::: Example
+
+The filter
+
+    . as [$x, {a: $y}] | $x, $y
+
+is equivalent to the filter
+
+    .[0]  as $x  |
+    .[1]  as $p1 |
+    $p1.a as $y  |
+    $x, $y
+
+Given the input `{a: 1, b: []}`, it returns twice `null`,
+because neither `.b` nor `.c[0]` are present in the input.
```

Comment:
You mean `.[0]` nor `.[1].a`?

Reply:
I changed the example, it was wrong. Thanks a lot for spotting this!